test/staking/staking.ts

```diff
@@ -0,0 +1,203 @@
+                expect((await stakedSYN.balanceOf((await signer.getAddress())))).to.eq(BigNumber.from(10).pow(18).mul(100))
+            })
+            // wait an hour
+            await increaseTimestamp(3600)
+            await stakedSYN.distribute()
+            await asyncForEach([owner, user1, user2, user3], async (signer) => {
+                expect((await stakedSYN.underlyingBalanceOf((await signer.getAddress())))).to.eq("1006500000000000000000")
+            })
+            await increaseTimestamp(3600)
+            await stakedSYN.distribute()
+            await asyncForEach([owner, user1, user2, user3], async (signer) => {
+                expect((await stakedSYN.underlyingBalanceOf((await signer.getAddress())))).to.eq("1906750000000000000000")
+            })
+            await increaseTimestamp(3600)
+            await stakedSYN.connect(owner).stake(BigNumber.from(10).pow(18).mul(100))
+            await asyncForEach([user1, user2, user3], async (signer) => {
+                expect((await stakedSYN.underlyingBalanceOf((await signer.getAddress())))).to.eq("2807000000000000000000")
+            })
+            expect((await stakedSYN.underlyingBalanceOf((await owner.getAddress())))).to.eq("2906999999999999999999")
+            await increaseTimestamp(3599)
+            await stakedSYN.connect(user3).undelegate(BigNumber.from(10).pow(18).mul(100))
+            await asyncForEach([user1, user2, user3], async (signer) => {
+                expect((await stakedSYN.underlyingBalanceOf((await signer.getAddress())))).to.eq("3699055084745762711864")
+            })
+            expect((await stakedSYN.underlyingBalanceOf((await owner.getAddress())))).to.eq("3830834745762711864406")
+
+            await increaseTimestamp(3600)
+            await stakedSYN.distribute()
+            expect((await stakedSYN.underlyingBalanceOf((await user3.getAddress())))).to.eq("3699055084745762711864")
+            await asyncForEach([user1, user2], async (signer) => {
+                expect((await stakedSYN.underlyingBalanceOf((await signer.getAddress())))).to.eq("4885301652050069718084")
+            })
+            expect((await stakedSYN.underlyingBalanceOf((await owner.getAddress())))).to.eq("5059341611154097851966")
+            await stakedSYN.connect(owner).stake(BigNumber.from(10).pow(18).mul(4711))
+
+            await increaseTimestamp(3600)
+            await stakedSYN.distribute()
+
+            await asyncForEach([user1, user2], async (signer) => {
+                expect((await stakedSYN.underlyingBalanceOf((await signer.getAddress())))).to.eq("5785813711574260203257")
+            })
+            expect((await stakedSYN.underlyingBalanceOf((await owner.getAddress())))).to.eq("11571317492105716881621")
+
+            // user3 can now unstake
+            await increaseTimestamp(604800)
+
+            await stakedSYN.connect(user3).unstake()
+            await expect((await SYN.balanceOf(user3Address))).to.eq("103599055084745762711864")
+
```

Comment:
Make all other stakers fully undelegate and check that they all are able to fully unstake (leaving the sSYN contract empty)